OpenGL-Core/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -46,6 +46,10 @@ namespace GLCore {
 			s_GLFWInitialized = true;
 		}
 
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+		glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+		glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
```

Comment:
Suggested addition of:
```c
glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
```
before the OpenGL profile hint line, because we're using 3.3 which for some OS (macOS) may require core context compatibility-mode enabled. This is due to Apple's move on starting to deprecate a bunch of APIs - OpenGL, OpenCL, etc. in favor of Metal.

I am aware that this MR is solely about Linux support, but just wanted to mention a point of consideration about GL context support:
> macOS: The OS only supports forward-compatible core profile contexts for OpenGL versions 3.2 and later. Before creating an OpenGL context of version 3.2 or later you must set the GLFW_OPENGL_FORWARD_COMPAT and GLFW_OPENGL_PROFILE hints accordingly. OpenGL 3.0 and 3.1 contexts are not supported at all on macOS.
More on this in the glfw official docs: https://www.glfw.org/docs/3.3/window_guide.html#GLFW_OPENGL_FORWARD_COMPAT_hint

I am only suggesting this because I continued the work on my Mac at some point and had to adjust a couple of things, though I've mostly completed adding a replacement Metal backend.

It is up to you and maybe @TheCherno if he'd like to have a look, but you can use the PLATFORM macro from the engine config and add it for mac platforms only, or not do anything but keep in mind for future reference.

Reply:
I'm not well versed on the specifics of OpenGL. I know that there are different OpenGL profiles but I don't know about the specifics of each. I held off on accepting the *Hints* PR into this branch until now since I rationalized that it is a minimum that can be just a starting point. Cherno plans to make videos on Apple support.

My philosophy behind Linux support is to increase the portability of this project. Apple support is good but the actions of that company are intended to close their platform in order to add proprietary value to their Intellectual Properties. Therefore anyone that buys an Apple shouldn't expect compatibility and I'm not going to dive in further to engineer around Apple. If *this* PR is accepted, a future PR could be submitted to modify the new base project.

Reply:
I see, I guess I just threw this out here for informative reasons. What you're saying makes sense. :)